examples/use_case_repository/use_case_repository/guides/pipes_cli_command.md

```diff
@@ -0,0 +1,88 @@
+---
```

Comment:
I think this template is a good start, but there are some changes I'd make:

- No need to have an **Overview** header, as the first paragraph just jumps right into that
- Add horizontal rules before each H2 heading to visually break up sections. Feel free to ignore this if this isn't the preferred style for `dagster.io` content.
- Move **What You'll Learn** up to come after the intro/overview, so users can decide immediately if they want to continue
- Move the **Expected Outcomes** content to come immediately after **Steps to Implement with Dagster** - this basically describes what users will do by following the steps, so we may as well tell them before they jump in
- Take individual implementation steps out of list format and put them into sections with H3 headings. I'm suggesting this for a few reasons:
   - Headings make the content easier to skim
   - Content is more visually appealing - IMO having lists with single items looks clunky, ex:
     ```
     1. **Step 1**
        - Do this thing
     2. **Step 2**
        - Do another thing
     ```
   - Markdown can get cranky about spacing, and keeping the spacing of code snippets in line with the list items that contain them is a PITA
- Move **Additional Resources** to come last - this may be personal preference, but having them come last makes me feel like I don't have to read through a section I may not care about to get to one I do care about, like **Next Steps**

Implementing all of the above would have this page look like the following:

````
# Running CLI Commands with Dagster Pipes

This guide demonstrates how to use Dagster Pipes to run a CLI command within a Dagster asset. This is useful for integrating non-Python workloads, such as Bash scripts or other command-line tools, into your Dagster data pipeline.

---

## What You’ll Learn

You will learn how to:

- Define a Dagster asset that invokes a CLI command.
- Use Dagster Pipes to manage subprocess execution.
- Capture and utilize the output of the CLI command within Dagster.

---

## Prerequisites

To follow the steps in this guide, you'll need:

- To have Dagster and the Dagster UI (`dagster-webserver`) installed. Refer to the [Installation guide](https://docs.dagster.io/getting-started/install) for more info.

---

## Steps to Implement with Dagster

By following these steps, you will have a Dagster asset that successfully runs a CLI command and logs its output. This allows you to integrate non-Python workloads into your Dagster data pipeline.

### Step 1: Define the CLI Command Script

Create a script that contains the CLI command you want to run. For example, create a file named `external_script.sh` with the following content:

```bash
#!/bin/bash
echo "Hello from CLI"
echo "My env var is: ${MY_ENV_VAR}"
```

### Step 2: Define the Dagster Asset

Define a Dagster asset that uses `PipesSubprocessClient` to run the CLI command. Include any necessary environment variables or additional parameters.

Save the following file to `dagster_pipes_cli.py`:

```python
import shutil

from dagster import AssetExecutionContext, Definitions, PipesSubprocessClient, asset

@asset
def cli_command_asset(
    context: AssetExecutionContext, pipes_subprocess_client: PipesSubprocessClient
):
    cmd = [shutil.which("bash"), "external_script.sh"]
    return pipes_subprocess_client.run(
        command=cmd,
        context=context,
        env={"MY_ENV_VAR": "example_value"},
    ).get_materialize_result()

defs = Definitions(
    assets=[cli_command_asset],
    resources={"pipes_subprocess_client": PipesSubprocessClient()},
)
```

### Step 3: Configure and Run the Asset

Ensure the script is executable and run the Dagster asset to see the output.

```bash
chmod +x external_script.sh
dagster dev -f path_to_your_dagster_file.py
```

---

## Troubleshooting

- **Permission Denied**: Ensure the script file has executable permissions using `chmod +x`.
- **Command Not Found**: Verify the command is available in the system's `PATH` or provide the full path to the command.

---

## Next Steps

Explore more advanced use cases with Dagster Pipes, such as integrating with other command-line tools or handling more complex workflows.

---

## Additional Resources

- [Dagster Pipes Documentation](https://docs.dagster.io/guides/dagster-pipes)
- [Dagster Installation Guide](https://docs.dagster.io/getting-started/install)
````

Reply:
Awesome thanks! Just updated both files with the new template.